app/src/main/java/de/danoeh/antennapod/dialog/FeedsApplyActionFragment.java

```diff
@@ -0,0 +1,365 @@
+import android.os.Bundle;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.ListView;
+
+import androidx.annotation.IdRes;
+import androidx.annotation.NonNull;
+import androidx.annotation.PluralsRes;
+import androidx.annotation.StringRes;
+import androidx.appcompat.app.AlertDialog;
+import androidx.appcompat.widget.Toolbar;
+
+import androidx.collection.ArrayMap;
+import androidx.fragment.app.Fragment;
+
+import com.google.android.material.snackbar.Snackbar;
+import com.leinardi.android.speeddial.SpeedDialView;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import de.danoeh.antennapod.R;
+import de.danoeh.antennapod.activity.MainActivity;
+import de.danoeh.antennapod.core.feed.Feed;
+import de.danoeh.antennapod.core.feed.FeedItem;
+import de.danoeh.antennapod.core.storage.DBWriter;
+import de.danoeh.antennapod.core.util.FeedItemPermutors;
+import de.danoeh.antennapod.core.util.FeedPermutors;
+import de.danoeh.antennapod.core.util.LongList;
+import de.danoeh.antennapod.core.util.FeedPermutors.SortOrder;
+import de.danoeh.antennapod.ui.common.ThemeUtils;
+import io.reactivex.Completable;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+
+@SuppressLint("NewApi")
+public class FeedsApplyActionFragment extends Fragment implements Toolbar.OnMenuItemClickListener {
+    private static final String TAG = "FeedsApplyActionFragmen";
+    public static final int ACTION_REMOVE_NEW_FLAGS = 1;
+    public static final int ACTION_MARK_AS_PLAYED = 2;
+    public static final int ACTION_REMOVE_FEED = 3;
+    public static final int ACTION_ALL = ACTION_REMOVE_NEW_FLAGS | ACTION_MARK_AS_PLAYED
+            | ACTION_REMOVE_FEED;
+
+
+    /**
+     * Specify an action (defined by #flag) 's UI bindings.
+     *
+     * Includes: the menu / action item and the actual logic
+     */
+    private static class ActionBinding {
+        int flag;
+        @IdRes
+        final int actionItemId;
+        @NonNull
+        final Runnable action;
+
+        ActionBinding(int flag, @IdRes int actionItemId, @NonNull Runnable action) {
+            this.flag = flag;
+            this.actionItemId = actionItemId;
+            this.action = action;
+        }
+    }
+
+    private final List<? extends ActionBinding> actionBindings;
+    private final Map<Long, Feed> idMap = new ArrayMap<>();
+//    private final List<FeedItem> episodes = new ArrayList<>();
+    private int actions;
+    private final List<String> titles = new ArrayList<>();
+    private final LongList checkedIds = new LongList();
+
+    private ListView mListView;
+    private ArrayAdapter<String> mAdapter;
+    private SpeedDialView mSpeedDialView;
+    private androidx.appcompat.widget.Toolbar toolbar;
```

Comment:
Doesn't need to be specified

```suggestion
    private Toolbar toolbar;
```